### PR TITLE
OHRM5X-1956: Develop Assign claim operation

### PR DESCRIPTION
**Pull request by DulsaraNethmin**

### CODE REVIEW

File: src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php [hunk 1]
```diff
@@ -116,6 +117,12 @@ public function create(): EndpointResult
         $claimRequest = new ClaimRequest();
         $empNumber = $this->getEmpNumber();
 
+        if (!is_null(
+            $this->getEmployeeService()->getEmployeeDao()->getEmployeeByEmpNumber($empNumber)
+                ->getEmployeeTerminationRecord()
```
Comment (ManulMax) [inline src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php:122]: test for deleted employees as well

File: src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php [hunk 2]
```diff
@@ -363,6 +379,20 @@ protected function setEmpNumbers(ClaimRequestSearchFilterParams $claimRequestSea
         }
     }
 
+    /**
+     * @param ClaimRequestSearchFilterParams $ClaimRequestSearchFilterParams
+     */
+    private function setEmpName(ClaimRequestSearchFilterParams $ClaimRequestSearchFilterParams)
+    {
+        $empName = $this->getRequestParams()->getStringOrNull(
```
Comment (ManulMax) [inline src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php:387]: $employeeName

File: src/plugins/orangehrmClaimPlugin/config/routes.yaml [hunk 3]
```diff
@@ -1,35 +1,35 @@
 apiv2_claim_events:
   path: /api/v2/claim/events
   controller: OrangeHRM\Core\Controller\Rest\V2\GenericRestController::handle
-  methods: [ POST, GET, DELETE ]
+  methods: [POST, GET, DELETE]
```
Comment (ManulMax) [inline src/plugins/orangehrmClaimPlugin/config/routes.yaml:4]: revert this

**Review by ManulMax — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmClaimPlugin/Dao/ClaimDao.php [hunk 4]
```diff
@@ -482,6 +482,13 @@ protected function getClaimRequestPaginator(ClaimRequestSearchFilterParams $myCl
                 ->setParameter('empNumbers', $myClaimRequestSearchFilterParams->getEmpNumbers());
         }
 
+        if (!is_null($myClaimRequestSearchFilterParams->getEmployeeName())) {
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Dao/ClaimDao.php:485]: @ManulMax please check this

File: installer/Migration/V5_5_0/Migration.php [hunk 5]
```diff
@@ -128,15 +128,15 @@ public function up(): void
                 'ohrm_claim_event',
                 ['id'],
                 'claimEventId',
-                ['onDelete' => 'RESTRICT']
+                ['onDelete' => 'CASCADE']
```
Comment (RajithaKumara) [inline installer/Migration/V5_5_0/Migration.php:131]: @ManulMax check 4.x behavior
seems `set null` is 4.x behavior.

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: installer/Migration/V5_5_0/Migration.php (hunk 5 above)
Comment (ManulMax) [reply installer/Migration/V5_5_0/Migration.php:131]: Checked with BA and said need to delete all claim request related to employee if we delete that employee.

File: src/plugins/orangehrmClaimPlugin/Dao/ClaimDao.php (hunk 4 above)
Comment (ManulMax) [reply src/plugins/orangehrmClaimPlugin/Dao/ClaimDao.php:485]: This one need to fix. Checked in leave, recruitment and timesheet employee search. Those showing invalid message if we try to search random word.

File: src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php [hunk 6]
```diff
@@ -115,6 +115,14 @@ public function create(): EndpointResult
     {
         $claimRequest = new ClaimRequest();
         $empNumber = $this->getEmpNumber();
+        $employee = $this->getEmployeeService()->getEmployeeDao()->getEmployeeByEmpNumber($empNumber);
+        if (is_null($employee)) {
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php:119]: I prefer
```suggestion
        if (!$employee instanceof Employee)) {
```
or
```suggestion
        if ($employee === null)) {
```

File: src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php [hunk 7]
```diff
@@ -347,7 +355,9 @@ protected function setEmpNumbers(ClaimRequestSearchFilterParams $claimRequestSea
         );
 
         if (!is_null($empNumber)) {
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php:357]: can `$empNumber` nullable?

File: src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php [hunk 8]
```diff
@@ -347,7 +355,9 @@ protected function setEmpNumbers(ClaimRequestSearchFilterParams $claimRequestSea
...
-            if (!$this->getEmployeeService()->getEmployeeDao()->getEmployeeByEmpNumber($empNumber) instanceof Employee) {
+            if (!$this->getEmployeeService()->getEmployeeDao()->getEmployeeByEmpNumber(
+                $empNumber
+            ) instanceof Employee) {
                 throw $this->getRecordNotFoundException();
```
Comment (RajithaKumara) [inline src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php:361]: seems record not found already handled above :thinking:

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php (hunk 7 above)
Comment (DulsaraNethmin) [reply src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php:357]: yes. it's a filterparam

File: src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php (hunk 8 above)
Comment (DulsaraNethmin) [reply src/plugins/orangehrmClaimPlugin/Api/EmployeeClaimRequestAPI.php:361]: here it's handling for the getAll query parameter

**Comment by RajithaKumara** [comment]
Closing as diff moved to #1692, https://github.com/orangehrm/orangehrm/pull/1692/commits/8cdc50ef3b19987c0fd22d9b52265aa67222e165

**Pull request closed without merging**